testdata/p4_16_samples_outputs/extract_struct-first.p4

```diff
@@ -0,0 +1,35 @@
+#include <core.p4>
+
+struct Tst {
+    bit<32> sdata;
+}
+
+header Header {
+    bit<32> data;
+    Tst     t;
+}
+
+parser p0(packet_in p, out Header h) {
+    bool b = true;
+    state start {
+        p.extract<Header>(h);
+        p.extract<Tst>(h.t);
```

Comment:
If you want to allow extraction of just structs perhaps you should write a more comprehensive test and some negative tests that check all kinds of structs.

Reply:
This file is getting removed. Very early on, during this work, the PR was changing extract behavior, but then the feedback provided ask to not change the behavior.

Reply:
Very early on, during this work, the PR was changing extract behavior, but then the feedback provided ask to not change the behavior.

Reply:
it looks like you have changed the behavior, since this extract accepts a header and does not report an error. So either you revert the behavior in the compiler or you add a test to check for this behavior.